src/avram/base_query_template.cr

```diff
@@ -66,6 +66,15 @@ class Avram::BaseQueryTemplate
                   foreign_key: primary_key_name
                 )
               )
+            {% elsif assoc[:relationship_type] == :has_one %}
+              join(
+                Avram::Join::{{ join_type.id }}.new(
+                  from: @@table_name,
+                  to: :{{ assoc[:table_name] }},
```

Comment:
This is singular, but needs to be plural for this to work. What's the best way to do that?

Reply:
Just for more info on this: The current failing spec mentions that relation `price` does not exist.  This is where `:price` gets set, but it should probably be more like `table_for(Price)`, or `"price".pluralize`?... This is where it gets a little tricky

Reply:
Do you know why it works in all the other spots? Maybe the table_name that gets set for `has_one` needs to be pluralized when set? 

Either way, I believe a way to get at this is to do `assoc[:type]::TABLE_NAME`. I'm not sure, but I think that would work

Reply:
You could go the `table_for({{ assoc[:type] }})` route but it would break if someone set a custom table name

Reply:
Here's where we set the `TABLE_NAME` constant for models: https://github.com/luckyframework/avram/blob/bbe0d2d1eab82ff9e9501c2d191af43bd94e2dc4/src/avram/model.cr#L132

Reply:
Oh and this is prob where you'd want to change it: https://github.com/luckyframework/avram/blob/bbe0d2d1eab82ff9e9501c2d191af43bd94e2dc4/src/avram/associations/has_one.cr#L20 so that way it gets the right table name from the start